ui/hooks/useGasFeeInputs.js

```diff
@@ -240,12 +248,16 @@ export function useGasFeeInputs(
   // When a user selects an estimate level, it will wipe out what they have
   // previously put in the inputs. This returns the inputs to the estimated
   // values at the level specified.
-  const setEstimateToUse = useCallback((estimateLevel) => {
-    setInternalEstimateToUse(estimateLevel);
-    setMaxFeePerGas(null);
-    setMaxPriorityFeePerGas(null);
-    setGasPrice(null);
-  }, []);
+  const setEstimateToUse = useCallback(
+    (estimateLevel) => {
+      setInternalEstimateToUse(estimateLevel);
```

Comment:
As per my other comments, we probably should not `setMaxFeePerGas(null);` in this function, unless there is a "gasLimit" too low error.

In that case, I think we should set the gasLimit to what is in the `txParams` instead of the minimum. `txParams.gas` should be considered the estimate of the gas. If there is no gas limit in the txParams, it can be set to the minimum.

Reply:
I'm ... having a hard time understanding what needs to be done here oO

Reply:
something like
```
if (gasErrors.gasLimit === GAS_FORM_ERRORS.GAS_LIMIT_OUT_OF_BOUNDS) {
    setGasLimit(hexToDecimal(tx.txParams.gas));
}
```

Reply:
Ok @danjm I think the confusion might have stemmed from you first mentioning `setMaxFeePerGas(null)` when I think you meant to refer to `setGasLimit(hexToDecimal(minimumGasLimit));`? So this change makes sense to me, but wondering if we want to fall back to the minimum here if the gasLimit on the transaction is itself too low?

Reply:
@adonesky1 That sounds like a good idea. With that change, I think this PR is good to go.